src/shell/Shell.cpp

```diff
@@ -237,6 +137,8 @@ static Trap::TrapResult executeWASM(Store* store, const std::string& filename, c
           (func (export "print_f64_f64") (param f64 f64))
         )
     */
+    Walrus::WASI wasi = Walrus::WASI();
+    wasi.fillWasiFuncTable();
```

Comment:
`WASI` instance and its function table are created whenever `executeWASM` called.
But can we allocate `WASI` once and share it in each `executeWASM` call?

Reply:
It was a rookie mistake on my part, but yes, we can. I moved it into main after the creation of `Store`. If you'd like we could add a flag so that wasi is only allocated if it needs to be used.